modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuildController.java

```diff
@@ -409,17 +418,31 @@ private void scheduleBuildIndex(
             long enlistmentConsistencyToken,
             HybridTimestamp initialOperationTimestamp
     ) {
+        ZonePartitionId zonePartitionId = new ZonePartitionId(zoneId, partitionId);
+        ZonePartitionResources resources = partitionReplicaLifecycleManager.zonePartitionResourcesOrNull(zonePartitionId);
+        if (resources == null) {
+            // Already stopped/destroyed, ignore.
+            return;
+        }
+
         MvPartitionStorage mvPartition = mvPartitionStorage(mvTableStorage, zoneId, tableId, partitionId);
 
         IndexStorage indexStorage = indexStorage(mvTableStorage, partitionId, indexDescriptor);
 
+        @Nullable ReplicaTableSegment segment = replicaTableSegment(zonePartitionId, tableId, resources, indexDescriptor);
```

Comment:
Minor nitpick, but it makes sense to move this code before you retrieve the storages